tmt/steps/execute/__init__.py

```diff
@@ -335,15 +353,24 @@ def check_result_file(self, test: "tmt.Test") -> List["tmt.Result"]:
         result = result_list[0].split("=")[1].strip()
 
         # Map the restraint result to the corresponding tmt value
+        actual_result = ResultOutcome.ERROR
+        note: Optional[str] = None
+
         try:
-            data.result = ResultOutcome(result.lower())
+            actual_result = ResultOutcome(result.lower())
         except ValueError:
             if result == 'SKIP':
-                data.result = ResultOutcome.INFO
+                actual_result = ResultOutcome.INFO
             else:
-                data.result = ResultOutcome.ERROR
-                data.note = f"invalid test result '{result}' in result file"
-        return [tmt.Result(data=data, test=test)]
+                actual_result = ResultOutcome.ERROR
```

Comment:
Not needed? Already set to `ERROR` on line 356.

Reply:
Correct. Refactored a bit in 7b113c75.